boa/src/syntax/ast/op.rs

```diff
@@ -158,25 +395,118 @@ impl Display for BitOp {
     }
 }
 
-/// A comparitive operation between 2 values
+/// A comparison operator compares its operands and returns a logical value based on whether the comparison is true.
+///
+/// The operands can be numerical, string, logical, or object values. Strings are compared based on standard
+/// lexicographical ordering, using Unicode values. In most cases, if the two operands are not of the same type,
+/// JavaScript attempts to convert them to an appropriate type for the comparison. This behavior generally results in
+/// comparing the operands numerically. The sole exceptions to type conversion within comparisons involve the `===` and `!==`
+/// operators, which perform strict equality and inequality comparisons. These operators do not attempt to convert the operands
+/// to compatible types before checking equality.
+///
+/// More information:
+///  - MDN documentation:
+/// <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Expressions_and_Operators#Comparison>
```

Comment:
Maybe we could add the "Testing and Comparison Operations" link as an spec link: https://tc39.es/ecma262/#sec-testing-and-comparison-operations

Reply:
That might be the best choice.